### ISSUE TITLE
Asking about inner and outer loop

### ISSUE THREAD

**Issue by tranquangchung** [dragen1860/MAML-Pytorch#64]
![image](https://user-images.githubusercontent.com/19748790/141765076-97262d4c-f1d9-468e-adce-6e16ba3d4fb2.png)

As I know, support set is training data and query set is testing data. And the query set uses for evaluating performance of model. However, in this code I realize that 
```
In Line 138 loss_q.backward() 
```
=> using query set (testing data) to update gradient.
1. You can explain in more detail here
2. Why not use loss of support set to update gradient, more specifically, loss.backward() (line 85 and 114)

Thanks
